.github/workflows/build.yml

```diff
@@ -1,4 +1,4 @@
-name: Build and test
+name: Build application
```

Comment:
I would prefer that the build step also invoke the test steps because they are so closely related

Reply:
I would prefer if the 'build' actions sole responsibility is to build the app and have a separate job for testing the app. We will not be able to merge pull requests into main if any of the pipelines fail either way. I believe this improves the readability of the actions and makes future refactoring easier.

Reply:
For me, it's an edge case that Flutter supports running tests without an explicit build. 

It would have been the ideal case that tests were run on one of the build artefacts we create. Now the risk is that an app could possibly be built differently than the variant where tests are run, thus this could lead to functionality not working on a particular build.

Anyhow the argument of separation is fine. Would it then make sense to have a wrapper workflow which calls both build and test?

Reply:
I'm not sure i see how the test build could differ from the actual build? The code is checked out and generated using the 'setup flutter environment' action for both builds. Anyhow i have made the test workflow run on pushes to the main branch

Reply:
We are building a specific Android and iOS variant. We don't have a test case for it yet but we already have functionality which are native to the build, thus a test case could potentially behave differently on iOS vs Android.

Reply:
The MobilePay integration reacting to the deep link. That's what my memory tells me. Really hope I'm remembering correctly 🙃